src/components/features/Forest/TreeAddForm/index.tsx

```diff
@@ -27,13 +52,28 @@ const TreeAddForm = () => {
 
 	const handleSubmitEditedTreeInfo = (event: React.FormEvent) => {
 		event.preventDefault();
+
 		console.log(treeName, selectedFruit);
+
+		if (treeId) {
+			treeUpdateMutation.mutate({ treeId: Number(treeId), name: treeName, fruitType: selectedFruit }); //
+		} else {
+			treePostMutation.mutate({ name: treeName, fruitType: selectedFruit });
+		}
 	};
 
 	const handleGoBackClick = () => {
 		navigate(-1);
 	};
 
+	useEffect(() => {
+		if (treeId) {
+			const targetTree = trees?.filter((tree) => tree?.id === Number(treeId))[0];
```

Comment:
```suggestion
		if (treeId && trees) {
			const targetTree = trees.filter((tree) => tree.id === Number(treeId))[0];
```

조건문에서 `trees` 가 있는지도 확인하면 그 안 로직을 조금 더 간단하게 만들수도 있을 것 같습니다!